source/scpp/extra/DSizeChecks.cpp

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+
+    Contains unittest functions used for size checks of the D glue layer.
+
+    Note: This is not part of Stellar SCP code.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+#include <vector>
+#include "xdrpp/marshal.h"
+#include "xdrpp/types.h"
+#include "xdr/Stellar-SCP.h"
+#include "xdr/Stellar-types.h"
+#include "scp/Slot.h"
+#include "crypto/ByteSlice.h"
+
+using namespace xdr;
+using namespace stellar;
+
+/// note: tried to use template but linking to it is a nightmare
```

Comment:
Care to elaborate ?

Reply:
C++ side:

```C++
template <typename T>
unsigned long long cppSizeOf ( T &value ) { return sizeof(value); }

// instantiate
template unsigned long long cppSizeOf<Value>(Value &value);
```

D side:

```D
extern(C++) ulong cppSizeOf (T)(ref T);
alias cppSizeOf!Value cppSizeOf_;
```

Error:

```
Undefined symbols for architecture x86_64:
  "unsigned long cppSizeOf<xdr::xvector<unsigned char, 4294967292u> >(xdr::xvector<unsigned char, 4294967292u>&)", referenced from:
      __D4scpd5tests8SizeTest17__unittest_L31_C1FZv in agora-test-application.o
```

I'm not sure how well D links with C++ templates. Maybe I was missing something?